app/controllers/people/omniauth_callbacks_controller.rb

```diff
@@ -1,18 +1,23 @@
 class People::OmniauthCallbacksController < Devise::OmniauthCallbacksController
   before_action :set_auth
   before_action :set_auth_mode
-  before_action :set_group_id
 
   def facebook
-    auth "facebook"
+    @service = "facebook"
+    send @auth_mode
```

Comment:
I find using send for dynamic method calling is pretty confusing here as it is non obvious what the possible auth_modes are here, and even what auth_mode is.

Reply:
but i changed my mind! revised. :)